dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env bash
+##
+## Copyright (c) 2025, Crash Override, Inc.
+##
+## This file is part of the Chalk project
+## (see https://crashoverride.com/docs/chalk)
+##
+
+set -euo pipefail
+
+# Enable debug mode only if DEBUG environment variable is set
+[[ ${DEBUG:-} == "true" ]] && set -x
+
+# Logger helpers with consistent formatting and timestamps
+log_with_level() {
+	local level="$1"
+	local color="$2"
+	local message="$3"
+	printf "\e[%sm[%s] [%s] %s\e[0m\n" "$color" "$(date -Iseconds)" "$level" "$message" >&2
```

Comment:
nit. `printf` I think is harder to read vs simple `echo`

for example in setup chalk:

https://github.com/crashappsec/setup-chalk-action/blob/44e40964dfd031b5d6b77ce5015a418c673ff649/setup.sh#L5-L50